### DISCUSSION TITLE
[v3] #[Reactive] properties 404 on delete

### DISCUSSION THREAD

**Discussion by MattiaMarchiorato** [livewire/livewire#6302] [7. Bugs (archived)]
Hello Devs!

following the nested components docs i tryed to set up a nested component with reactive attribute to count a todos
https://livewire.laravel.com/docs/nesting

```
<?php
 
namespace App\Livewire;
 
use Livewire\Attributes\Reactive;
use Livewire\Component;
use App\Models\Todo;
 
class TodoCount extends Component
{
    #[Reactive] 
    public $todos;
 
    public function render()
    {
        return view('livewire.todo-count', [
            'count' => $this->todos->count(),
        ]);
    }
}
```

when i call the remove method form the TodoList component i receive a 404

```
#[On('remove-todo')] 
public function remove(Todo $todo)
{  
    $todo->delete();
}
```

Seems the newsted component/property is not updated and not reactive on delete.
If i remove the  `#[Reactive] ` attribute and use key on the nested component include seems working fine.

Thanks for help

MM

**Comment by MattiaMarchiorato** [comment]
Any news?

**Reply by daronspence** [reply]
I just ran into this today as well. Stripped it all down and when a nested child has a reactive prop and is deleted, the app was throwing a 404.

In my case, it was a boolean.

### Nested Child
```php
<?php

namespace App\Livewire\Pokers;

use App\Models\Poker;
use Livewire\Attributes\Reactive;
use Livewire\Component;

class ShowPokerTest extends Component
{
    public Poker $poker;

    #[Reactive]
    public ?bool $spectating = null;

    public function render()
    {
        return <<<'BLADE'
            <div>
                <div>{{ $this->poker->name }}</div>
                <button wire:click="$parent.deletePoker({{ $this->poker->id }})">
                    <span>Delete</span>
                </button>
            </div>
        BLADE;
    }
}
```

### Parent
```php
<?php

namespace App\Livewire\Sessions;

use App\Events\SessionUpdated;
use App\Models\Poker;
use App\Models\Session;
use Livewire\Attributes\Computed;
use Livewire\Component;

class ShowSessionPage extends Component
{
    public Session $session;

    public ?bool $spectating = null;

    public function mount()
    {
        $this->spectating =
            currentUser()->activeSession->first()->pivot->spectating ?? false;
    }

    public function boot()
    {
        if (! is_null($this->spectating)) {
            currentUser()->setActiveSession($this->session, $this->spectating);
        }
    }

    public function deletePoker($pokerId)
    {
        $poker = Poker::find($pokerId);
        $this->authorize('delete', $poker);
        $poker->delete();
        SessionUpdated::dispatch();
    }

    #[Computed]
    public function pokers()
    {
        return $this->session->refresh()->pokers()->get()->reverse();
    }

    public function render()
    {
        return <<<'BLADE'
        <div>
            {{ $session->name }}
            <div>
                @foreach ($this->pokers as $poker)
                    <livewire:pokers.show-poker-test
                        :$poker
                        :$spectating
                        :key="$poker->id"
                    />
                @endforeach
            </div>
        </div>
        BLADE;
    }
}
```

If I remove the `#[Reactive]` in the child, it works fine and I can delete the child through the parent model with no issues.

**Reply by MattiaMarchiorato** [reply]
Yes, is the exact same issue, but  no one cares about it 😂😩

**Reply by daronspence** [reply]
@Mattia-Marchiorato are you comfortable writing a Test for this? I've heard that the best way to get things fixed in Livewire is with failing tests. I don't have time to do it atm, (or which feature this belongs to) but maybe in another week or two I will have some time to try.

**Reply by MattiaMarchiorato** [reply]
@daronspence at the moment we are a bit busy on a project, maybe in 2 weeks we can work on it, please keep in touch.

**Comment by dairidong** [comment]
Have same problems, any news?

**Comment by MomoDerDodo** [comment]
Still a problem on 30.06.2024

**Comment by bernig** [comment]
The issue persists. I had to temporarily rely on dispatching an event to update the property until a solution is found.

**Comment by matthewhaworth** [comment]
I think I have a very similar issue to this with Modelable instead of Reactive, I've created my own discussion here: https://github.com/livewire/livewire/discussions/9126

**Comment by thomas-alrek** [comment]
I'm experiencing the same issue. Is there any workaround for this?

**Reply by lrljoe** [reply]
Please share your parent and child component, as the above example from another poster isn't complete/relevant for you.
